packages/updatePrescriptionStatus/src/updatePrescriptionStatus.ts

```diff
@@ -284,6 +286,33 @@ function response(statusCode: number, responseEntries: Array<BundleEntry>) {
   }
 }
 
+async function logTransitions(dataItems: Array<DataItem>, logger: Logger): Promise<void> {
+  for (const dataItem of dataItems) {
+    try {
+      const previousItem = await getPreviousItem(dataItem)
+      if (previousItem) {
+        const newDate = new Date(dataItem.LastModified)
+        const previousDate = new Date(previousItem.LastModified)
+        logger.info("Transitioning item status.", {
+          prescriptionID: dataItem.PrescriptionID,
+          lineItemID: dataItem.LineItemID,
+          nhsNumber: dataItem.PatientNHSNumber,
+          pharmacyODSCode: dataItem.PharmacyODSCode,
+          applicationName: dataItem.ApplicationName,
+          when: dataItem.LastModified,
+          interval: (newDate.valueOf() - previousDate.valueOf()) / 1000,
+          newStatus: dataItem.Status,
+          previousStatus: previousItem.Status,
+          newTerminalStatus: dataItem.TerminalStatus,
+          previousTerminalStatus: previousItem.TerminalStatus
+        })
+      }
+    } catch (e) {
+      logger.error("Error logging transition.", {taskID: dataItem.TaskID, error: e})
+    }
```

Comment:
Have we considered how we will account for updates that fail logging in the final dataset?
I imagine they'll be a very small proportion, and it's probably not worth erroring the transaction over, so it's not super important if we're happy to ignore them

Reply:
We should be watching for these errors and fix if they happen, the reason for swallowing the exception is we don't want to break anything with this new code as it could be a deviation from what the supplier tested with.